src/services/apple_music.js

```diff
@@ -39,12 +39,21 @@ export default class AppleMusic {
     if (typeof config !== 'object') throw new Error(`[AppleMusic] Please define a configuration as an object`);
     if (!config.developerToken)
       throw new Error(`[AppleMusic] Please define [developerToken] as a property within the configuration`);
-    this.#store.core = new Client({developerToken: config.developerToken});
-    let custom_head = {headers: {origin: 'https://music.apple.com'}};
-    this.#store.core.albums.axiosInstance = this.#store.core.albums.axiosInstance.create(custom_head);
-    this.#store.core.artists.axiosInstance = this.#store.core.artists.axiosInstance.create(custom_head);
-    this.#store.core.playlists.axiosInstance = this.#store.core.playlists.axiosInstance.create(custom_head);
-    this.#store.core.songs.axiosInstance = this.#store.core.songs.axiosInstance.create(custom_head);
+      
+    this.#store.core = new Client({ developerToken: config.developerToken });
+
+    function augmentAxiosInstance(instance) {
+      instance.interceptors.request.use((conf) => {
+        conf.headers.origin = 'https://music.apple.com';
+        return conf;
+      });
+    }
+
+    augmentAxiosInstance(this.#store.core.albums.axiosInstance);
+    augmentAxiosInstance(this.#store.core.artists.axiosInstance);
+    augmentAxiosInstance(this.#store.core.playlists.axiosInstance);
+    augmentAxiosInstance(this.#store.core.songs.axiosInstance);
```

Comment:
Even though this issue has already been fixed – https://github.com/miraclx/freyr-js/pull/215 (issue with yarn. not the freyr)

I do like this solution.

Although lets use something of this fashion:

```suggestion
    for (let instance of [
      this.#store.core.albums,
      this.#store.core.artists,
      this.#store.core.playlists,
      this.#store.core.songs,
    ])
      instance.axiosInstance.interceptors.request.use(
        conf => ((conf.headers.origin = 'https://music.apple.com'), conf),
      );
```